crates/iceberg/src/scan.rs

```diff
@@ -159,6 +190,21 @@ impl TableScan {
             while let Some(entry) = entries.next().await {
                 let manifest = entry.load_manifest(&file_io).await?;
 
+                // If this scan has a filter, check the partition evaluator cache for an existing
+                // PartitionEvaluator that matches this manifest's partition spec ID.
+                // Use one from the cache if there is one. If not, create one, put it in
+                // the cache, and take a reference to it.
+                if let Some(filter) = self.filter.as_ref() {
+                    let partition_evaluator = partition_evaluator_cache
+                            .entry(manifest.partition_spec_id())
+                            .or_insert_with_key(|key| self.create_partition_evaluator(key, filter));
+
+                    // reject any manifest files whose partition values don't match the filter.
+                    if !partition_evaluator.filter_manifest_file(&entry) {
```

Comment:
If I understood correctly; the `ManifestEval` is also applied here? I think this should be done before loading each manifest file by evaluating the `partitions: Vec<FieldSummary>` of `ManifestList`.

Reply:
Good point - fixed.